dialogue-core/src/main/java/com/palantir/dialogue/core/DialogueNodeSelectionStrategy.java

```diff
@@ -50,7 +51,7 @@ static List<DialogueNodeSelectionStrategy> fromHeader(String header) {
      * which users can't normally configure.
      */
     private static DialogueNodeSelectionStrategy safeValueOf(String string) {
-        String uppercaseString = string.toUpperCase();
+        String uppercaseString = string.toUpperCase(Locale.ROOT);
```

Comment:
should we bump `DefaultLocale` to `error` in `build.gradle` line 89?
```
            options.errorprone.error('DefaultLocale')
```

An alternative to avoid the potential allocation which would fail fast based on length check as opposed to the switch `hashCode`, then do case insensitive `regionMatches`:

```
        if (string.equalsIgnoreCase("PIN_UNTIL_ERROR")) {
            return PIN_UNTIL_ERROR;
        } else if (string.equalsIgnoreCase("PIN_UNTIL_ERROR_WITHOUT_RESHUFFLE")) {
            return PIN_UNTIL_ERROR_WITHOUT_RESHUFFLE;
        } else if (string.equalsIgnoreCase("BALANCED")) {
            return BALANCED;
        }
```

Reply:
:+1: Updated with equalsIgnoreCase, and enabled DefaultLocale at warn (which will fail builds due to `-Werror`)